week4/client/posting-app/src/pages/PostDetail.jsx

```diff
@@ -2,51 +2,64 @@ import React, { useState, useEffect } from "react";
 import { useParams } from "react-router-dom";
 import { Container, Card } from "react-bootstrap";
 import { formatDistanceToNow } from "date-fns";
-import { useDocumentTitle } from "../utils/utils";
+import Comment from "../components/Comment";
 
 function PostDetail() {
-  useDocumentTitle("Bloggy Post Details");
-
   const { postId } = useParams();
   const [post, setPost] = useState(null);
+  const [comments, setComments] = useState([]);
 
   useEffect(() => {
-    const fetchPost = async () => {
+    const fetchPostAndComments = async () => {
       try {
-        const response = await fetch(
+        const postResponse = await fetch(
           `${process.env.REACT_APP_API_URL}/api/posts/${postId}`
         );
-        if (response.ok) {
-          const data = await response.json();
-          setPost(data);
-        } else {
-          console.error("Failed to fetch post:", response.status);
-        }
+        const postData = postResponse.ok ? await postResponse.json() : null;
+        setPost(postData);
+
+        const commentsResponse = await fetch(
+          `${process.env.REACT_APP_API_URL}/api/posts/${postId}/comments`
+        );
+        const commentsData = commentsResponse.ok
+          ? await commentsResponse.json()
+          : [];
+        setComments(commentsData);
       } catch (error) {
-        console.error("Error fetching post:", error);
+        console.error("Error fetching data:", error);
       }
     };
```

Comment:
This method does two things, which is not recommended. By creating two separate methods for fetching posts and comments, you can then utilize these methods within Promise.all inside the fetchPostAndComments function to execute the API calls for posts and comments concurrently. This facilitates the loading of both resources without the need for one to wait for the other, thereby reducing the overall load time. Once you've established two separate helper functions, you can employ Promise.all within the fetchPostAndComments function in the following manner:

`const [postData, commentsData] = await Promise.all([fetchPost(), fetchComments()]);`